File: tests/code/corpus/langs/zig/registry.zig
const shapes = @import("shapes.zig");

const Registry = struct {
    count: usize,

    pub fn seed(self: *Registry) void {
        self.count = 1;
    }
};

pub fn emptyRegistry() Registry {
    return Registry{ .count = 0 };
}

test "registry starts empty" {
    const r = emptyRegistry();
    try std.testing.expect(r.count == 0);
}

File: tests/code/corpus/langs/rust/registry.rs
use crate::shapes::make_circle;

pub struct Registry {
    pub count: usize,
}

impl Registry {
    pub fn seed(&mut self) {
        make_circle(1.0);
        self.count = 1;
    }
}

pub fn empty_registry() -> Registry {
    Registry { count: 0 }
}
